File: src/main/kotlin/dev/jefy/connectpro/portfolio/applicaion/query/PortfolioQuery.java
package dev.jefy.connectpro.portfolio.applicaion.query;

import dev.jefy.connectpro.portfolio.applicaion.dtos.JobPostResponse;
import dev.jefy.connectpro.portfolio.applicaion.dtos.PortfolioResponse;
import dev.jefy.connectpro.portfolio.applicaion.dtos.ServiceResponse;
import dev.jefy.connectpro.portfolio.domain.vo.JobPostId;
import dev.jefy.connectpro.portfolio.domain.vo.PortfolioId;
import dev.jefy.connectpro.portfolio.domain.vo.ServiceId;

/**
 * @author Jôph Yamba
 */
public interface PortfolioQuery {
    PortfolioResponse get(PortfolioId portfolioId);
    JobPostResponse getJopPost(JobPostId jobPostId);
    ServiceResponse getService(ServiceId serviceId );
}

File: src/main/kotlin/dev/jefy/connectpro/portfolio/application/query/PortfolioQuery.kt
package dev.jefy.connectpro.portfolio.application.query

import dev.jefy.connectpro.marketplace.application.dtos.JobPostListingResponse
import dev.jefy.connectpro.marketplace.application.dtos.ServiceListingResponse
import dev.jefy.connectpro.portfolio.application.dtos.JobPostResponse
import dev.jefy.connectpro.portfolio.application.dtos.PortfolioResponse
import dev.jefy.connectpro.portfolio.application.dtos.ServiceResponse
import dev.jefy.connectpro.portfolio.domain.vo.JobPostId
import dev.jefy.connectpro.portfolio.domain.vo.PortfolioId
import dev.jefy.connectpro.portfolio.domain.vo.ServiceId

interface PortfolioQuery {
    fun get(portfolioId: PortfolioId): PortfolioResponse
    fun getJobPost(jobPostId: JobPostId): JobPostResponse
    fun getAllJobPost(portfolioId: PortfolioId): List<JobPostListingResponse>
    fun getService(serviceId: ServiceId): ServiceResponse
    fun getAllService(portfolioId: PortfolioId): List<ServiceListingResponse>
}